activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/security/TaskLookupRestrictionService.java

```diff
@@ -0,0 +1,105 @@
+package org.activiti.cloud.services.security;
+
+import com.querydsl.core.types.Predicate;
+import com.querydsl.core.types.dsl.BooleanExpression;
+import org.activiti.cloud.services.query.model.QTask;
+import org.activiti.cloud.services.query.model.QVariable;
+import org.activiti.engine.UserGroupLookupProxy;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class TaskLookupRestrictionService {
+
+    @Autowired(required = false)
```

Comment:
Prefer constructor injection.

Reply:
Generally I do too with optional attributes it's not so great. These can be handled with the Optional class as per https://stackoverflow.com/questions/42135102/how-to-set-autowired-constructor-params-as-required-false-individually but it is not so readable and doesn't work with @InjectMocks like in https://github.com/Activiti/activiti-cloud-query-service/blob/69c537861ad48753d66712130f51e110fd8990c8/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/test/java/org/activit/cloud/services/security/SecurityPoliciesApplicationServiceTest.java#L37 so I'd then have to change that test class to use setters to inject the mocks.

Reply:
👍